sw/CMakeLists.txt

```diff
@@ -377,6 +440,7 @@ SET(CMAKE_EXE_LINKER_FLAGS  "-T ${LINKER_SCRIPT}  \
                              -Wl,-Map=${MAINFILE}.map \
                              -L ${RISCV}/${COMPILER_PREFIX}elf/lib \
                              -lc -lm -lgcc -flto \
+                             -fpermissive -fno-rtti -fno-exceptions -fno-threadsafe-statics  -fno-unwind-tables  \
```

Comment:
"-fpermissive" is used because there are multiple occasions where C code is causing errors in C++ (mostly type conversions). By using this, it allows the code to compile.
"-fno-unwind-tables" was used allong with "-fno-exceptions" to remove try-catch features, but according to newer gcc documentation it is no longer required, so I'll remove it.

By adding C++ as a feature, any new addition of device or library files, must be compatible with C++. So if C++ is added in the main branch, it needs to be taken into consideration when adding new code.